### PR TITLE
Remote API documentation

### PR DESCRIPTION
**Pull request by real2two**
# Remote API documentation

I decided start documenting Pterodactyl Panel's remote API routes. 

Quick side note: I might be missing some optional parameters Pterodactyl Panel sends to Wings on my documentation, but I didn't include them because Wings never uses them.

## Todo

- [x] Explain headers in README.md.
- [x] Replace 'example body's with actual example values, instead of the type.
- [x] Responses, including errors.
- [x] Provide example response models. https://github.com/devnote-dev/ptero-notes/issues/9 (edit: besides one route I couldn't figure out.)

### CODE REVIEW

**Comment by devnote-dev** [comment]
Please mark the PR as a draft then update it when it's ready for review.

**Comment by real2two** [comment]
I'm going to note that I might be missing some optional parameters/body on some routes. I'll try to fix anything I can find, but I can't guarantee anything.

**Comment by devnote-dev** [comment]
That's fine, when it's ready for review I'll go through the whole API.

**Comment by real2two** [comment]
I couldn't find this remote API route in Pterodactyl Panel's `develop` branch, which is pretty interesting.

`POST /servers/:server/archive` - The function's name is "SetArchiveStatus".
https://github.com/pterodactyl/wings/blob/bd26d6eefddcdbdc14779ae61b1ff126820241da/remote/servers.go#L104

**Comment by real2two** [comment]
I gave up response examples for `GET /api/remote/backups/:backup`, because I couldn't figure out how to get S3 to properly work on Pterodactyl Panel.

I decided to leave the response example as "WIP.", but everything else is complete, unless there are any small issues I need to change.

**Comment by NaysKutzu** [comment]
Everything you added works thanks you saved my ass @real2two

File: pterodactyl/remote/server_backups.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+### `GET /api/remote/backups/:backup`
+
+Returns the required presigned urls to upload a backup to S3 cloud storage.
+
+### Parameters
+
+| Name      | Visibility | Description                        |
+| --------- | ---------- | ---------------------------------- |
+| size      | optional   | The backup file size.              |
```
Comment (devnote-dev) [inline pterodactyl/remote/server_backups.md]:
```suggestion
| size      | required   | The backup file size.              |
```

File: pterodactyl/remote/server_backups.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+| Code | Description                                                    |
+| ---- | -------------------------------------------------------------- |
+| 200  | The response was successful.                                   |
+| 400  | The configured backup adapter is not an S3 compatible adapter. |
+| 404  | The backup was not found.                                      |
+| 409  | The backup is already in completed state.                      |
+
+### Response examples
+
+WIP.
```
Comment (devnote-dev) [inline pterodactyl/remote/server_backups.md]: Better to not include it for now.

```suggestion
```

File: pterodactyl/remote/server_backups.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+Sources
+
+- [app/Http/Controllers/Api/Remote/Backups/BackupRemoteUpl2oadController.php#L33](https://github.com/pterodactyl/panel/blob/v1.11.3/app/Http/Controllers/Api/Remote/Backups/BackupRemoteUploadController.php#L33)
+
+---
```
Comment (devnote-dev) [inline pterodactyl/remote/server_backups.md]:
```suggestion
```

File: pterodactyl/remote/server_backups.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+### `POST /api/remote/backups/:backup`
+
+Handles updating the state of a backup.
+
+### Body
+
+| Name  | Visibility | Type   | Description                                                                                            |
+| ----- | ---------- | ------ | ------------------------------------------------------------------------------------------------------ |
+| data  | required   | object | An object containing the checksum, checksum type, backup size, success state, and parts of the backup. |
```
Comment (devnote-dev) [inline pterodactyl/remote/server_backups.md]: The actual fields are in [ReportBackupCompleteRequest.php](https://github.com/pterodactyl/panel/blob/b23f3114e4cb71406e6de17d0dca0d88205f56c1/app/Http/Requests/Api/Remote/ReportBackupCompleteRequest.php).

File: pterodactyl/remote/server_installation.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
...N}-${FORGE_VERSION}.jar\r\n  fi\r\nfi\r\n\r\n#Adding .jar when not eding by SERVER_JARFILE\r\nif [[ ! $SERV\njava -jar installer.jar --installServer || { echo -e \"\\nInstall failed using Forge version ${FORGE_VERSION} and Minecraft version ${MINECRAFT_VERSION}.\\nShould you be using unlimited memory value of 0, make sure to increase the default install resource limits in the Wings config or specify exact allocated memory in the server Build Configuration instead of 0! \\nOtherwise, the Forge installer will not have enough memory.\"; exit 4; }\r\n\r\n# Check if we need a symlink for 1.17+ Forge JPMS args\r\nif [[ $MC_VERSION =~ ^1\\.(17|18|19|20|21|22|23) || $FORGE_VERSION =~ ^1\\.(17|18|19|20|21|22|23) ]]; then\r\n  unix_args\r\n\r\n# Check if someone has set MC to latest but overwrote it with older Forge version, otherwise we would have false positives\r\nelif [[ $MC_VERSION == \"latest\" && $FORGE_VERSION =~ ^1\\.(17|18|19|20|21|22|23) ]]; then\r\n  unix_args\r\nelse\r\n  # For versions below 1.17 that ship with jar\r\n  mv $FORGE_JAR $SERVER_JARFILE\r\nfi\r\n\r\necho -e \"Deleting installer.jar file.\\n\"\r\nrm -rf installer.jar\r\necho -e \"Installation process is completed\""
```
Comment (devnote-dev) [inline pterodactyl/remote/server_installation.md]: Please shorten this...

File: pterodactyl/remote/servers.md [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+server:startup.image
+server:subuser.create
+server:subuser.delete
+server:subuser.update
+server:task.create
+server:task.delete
+server:task.update
+
+* If there are any missing events, feel free to open a pull request.
+```
```
Comment (devnote-dev) [inline pterodactyl/remote/servers.md]: I feel like this should be placed in a different section as activities are enough to be their own topic.

File: pterodactyl/remote/sftp.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
+### `POST /api/remote/sftp/auth`
+
+Authenticate a set of credentials and return the associated server details for a SFTP connection on the daemon. This supports both public key and password based credentials.
+
+### Body
+
+| Name           | Visibility | Type   | Description                                                                                       |
+| -------------- | ---------- | ------ | ------------------------------------------------------------------------------------------------- |
+| type           | required   | string | Determine if the password should check authentication with either the "password" or "public_key". |
```
Comment (devnote-dev) [inline pterodactyl/remote/sftp.md]:
```suggestion
| type           | optional   | string | Determine if the password should check authentication with either the "password" or "public_key". |
```

File: pterodactyl/remote/sftp.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+| username       | required   | string | The username of the user attempting to login.                                                     |
+| password       | required   | string | The password of the user attempting to login.                                                     |
+| ip             | optional   | string | The remote address of the connection.                                                             |
+| session_id     | optional   | []byte | The connection's session id.                                                                      |
+| client_version | optional   | []byte | The connection's client version.                                                                  |
```
Comment (devnote-dev) [inline pterodactyl/remote/sftp.md]: These are not valid fields, see [SftpAuthenticationFormRequest.php](https://github.com/pterodactyl/panel/blob/b23f3114e4cb71406e6de17d0dca0d88205f56c1/app/Http/Requests/Api/Remote/SftpAuthenticationFormRequest.php).

**Review by devnote-dev — CHANGES_REQUESTED** [review]

File: pterodactyl/remote/sftp.md (hunk 8 above)
Comment (real2two) [reply pterodactyl/remote/sftp.md]: opps! I was taking the types from wings, and sometimes there are unnecessary request bodies.

File: pterodactyl/remote/servers.md (hunk 6 above)
Comment (real2two) [reply pterodactyl/remote/servers.md]: I'll add an `activity_events.md` later, unless you want me to name the file something else.

File: pterodactyl/remote/server_backups.md (hunk 4 above)
Comment (real2two) [reply pterodactyl/remote/server_backups.md]: I don't know what `etag` and `part_number` does, but I committed "???" as a placeholder.

**Comment by real2two** [comment]
Please read this:
- I placed events in `activity_events.md`. I didn't provide any descriptions for each event.
- I have no idea what `data.parts[].etag` and `data.parts[].part_number` does in [`POST /api/remote/backups/:backup`](https://github.com/real2two/ptero-notes/blob/remote/pterodactyl/remote/server_backups.md#post-apiremotebackupsbackup), so I added "???" as a placeholder description, so you might want to check that out.

File: pterodactyl/remote/server_backups.md (hunk 4 above)
Comment (real2two) [reply pterodactyl/remote/server_backups.md]: I'll resolve this anyways for now.

**Review by devnote-dev — APPROVED** [review]

**Pull request merged**